splunk-otel-android/src/test/java/com/splunk/rum/SplunkSpanDataModifierTest.java

```diff
@@ -195,4 +195,95 @@ public void shouldCopySelectedResourceAttributes() {
                         equalTo(SplunkRum.APP_NAME_KEY, "myApp"),
                         equalTo(SplunkRum.RUM_VERSION_KEY, "1.0.0"));
     }
+
+    @Test
+    public void shouldIgnoreReactIdsIfReactNativeSupportIsDisabled() {
+        SpanContext spanContext =
+                SpanContext.create(
+                        "00000000000000000000000000000123",
+                        "0000000000000456",
+                        TraceFlags.getSampled(),
+                        TraceState.getDefault());
+
+        SpanData original =
+                TestSpanData.builder()
+                        .setSpanContext(spanContext)
+                        .setName("SplunkRumSpan")
+                        .setKind(SpanKind.CLIENT)
+                        .setStatus(StatusData.unset())
+                        .setStartEpochNanos(12345)
+                        .setEndEpochNanos(67890)
+                        .setHasEnded(true)
+                        .setAttributes(
+                                Attributes.builder()
+                                        .put(
+                                                REACT_NATIVE_TRACE_ID_KEY,
+                                                "99999999999999999999999999999999")
+                                        .put(REACT_NATIVE_SPAN_ID_KEY, "8888888888888888")
+                                        .build())
+                        .build();
+
+        CompletableResultCode exportResult = CompletableResultCode.ofSuccess();
+        when(delegate.export(exportedSpansCaptor.capture())).thenReturn(exportResult);
+
+        SpanExporter underTest = new SplunkSpanDataModifier(delegate, false);
+        CompletableResultCode actual = underTest.export(singleton(original));
+
+        assertThat(actual).isSameAs(exportResult);
+
+        Collection<SpanData> exportedSpans = exportedSpansCaptor.getValue();
+        assertThat(exportedSpans).hasSize(1);
+        assertThat(exportedSpans.iterator().next())
+                .hasName("SplunkRumSpan")
+                .hasTraceId(spanContext.getTraceId())
+                .hasSpanId(spanContext.getSpanId())
+                .hasAttributesSatisfyingExactly(
+                        equalTo(SPLUNK_OPERATION_KEY, "SplunkRumSpan"),
+                        equalTo(REACT_NATIVE_TRACE_ID_KEY, "99999999999999999999999999999999"),
+                        equalTo(REACT_NATIVE_SPAN_ID_KEY, "8888888888888888"));
```

Comment:
We still want to carry these forward even when reactive native support isn't active? I mean, I get that they're just attributes, but still....?

Not sure if this is a pro or con -- one the one hand, seeing these in the UI as attributes might at least help someone to understand where the span came from, but the downside is that it might just cause confusion. I dunno. Feels like a leaky implementation detail.

Reply:
Honestly, no idea. My idea was to simply treat them as "normal" attributes in case RN support is off; but always dropping them is also not that bad of a choice.